ddclient.in

```diff
@@ -2155,6 +2168,171 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## curl_cmd() function to execute system curl command
+######################################################################
+sub curl_cmd {
+    my @params = @_;
+    my $tmpfile;
+    my $tfh;
+    my $system_curl = subst_var('@syscurlcmd@', '/usr/bin/curl');
+    my %curl_codes = (  ## Subset of error codes from https://curl.haxx.se/docs/manpage.html
+         2 => "Failed to initialize. (Most likely a bug in ddclient, please open issue at https://github.com/ddclient/ddclient)",
+         3 => "URL malformed. The syntax was not correct",
+         5 => "Couldn't resolve proxy. The given proxy host could not be resolved.",
+         6 => "Couldn't resolve host. The given remote host was not resolved.",
+         7 => "Failed to connect to host.",
+        22 => "HTTP page not retrieved. The requested url was not found or returned another error.",
+        28 => "Operation timeout. The specified time-out period was reached according to the conditions.",
+        35 => "SSL connect error. The SSL handshaking failed.",
+        47 => "Too many redirects. When following redirects, curl hit the maximum amount.",
+        51 => "The peer's SSL certificate or SSH MD5 fingerprint was not OK.",
+        58 => "Problem with the local certificate.",
+        60 => "Peer certificate cannot be authenticated with known CA certificates.",
+        67 => "The user name, password, or similar was not accepted and curl failed to log in.",
+        77 => "Problem with reading the SSL CA cert (path? access rights?).",
+        78 => "The resource referenced in the URL does not exist.",
+       127 => "You requested network access with curl but $system_curl was not found",
+    );
+
+    return '' if (scalar(@params) == 0); ## no parameters provided
+
+    # debug("CURL: %s\r\n",join("\r\n      ",@params)); ## Uncomment only if required as may expose credentials
```

Comment:
It is NOT okay to include credentials in debug, at least not in my book.  It is very common for people to copy/paste their debug output in bug reports and not all credentials are obvious that that is what they are.  So I do not think we should uncomment this. 

We could delete it, but do you really feel that strongly?  I have found it very useful while developing to be able to easily uncomment this.

Reply:
> It is NOT okay to include credentials in debug

There's a trade-off. It's not always possible to censor/omit sensitive information, but we do need debug information on occasion. I think we should address this by using `verbose()` more often so that `verbose=yes debug=no` logs contain enough information most of the time.

> We could delete it, but do you really feel that strongly?

Yes. Commented code tends to rot over time, it impairs readability, and it makes the surrounding code harder to maintain. If the code is frequently used then it should instead be guarded by a flag/option (such as `-debug`). If it's not frequently used then it's no big deal to recreate it if/when needed.

Reply:
I will delete the line